src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `P _ᗮ` for `orthogonal_projection _ᗮ`.
+
+We let `V` be a finite-dimensional inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P`U := (orthogonal_projection U)
+local notation `P`U`ᗮ` := (orthogonal_projection Uᗮ)
+
+/-- `U` is `T` invariant is equivalent to `T(U) ⊆ U` -/
+lemma U_is_T_invariant_def (U : submodule ℂ V) (T : V →L[ℂ] V) :
+ (∀ u : V, u ∈ U → T u ∈ U) ↔ (T '' U ⊆ U)
+  := by { simp only [set.image_subset_iff],
+          exact ⟨ λ h, λ x hx, by simp only [set.mem_preimage]; exact h x hx,
+                  λ h u h_1, by apply h h_1 ⟩, }
```

Comment:
This lemma is just `set.image_subset_iff` in reverse,

Reply:
Whoops! Just noticed lol. Removed that and put it as a definition.